### PR TITLE
change reduced scan query to merge all 4 data sources

### PR DESCRIPTION
**Pull request by ohnorobo**
New tables
- `firehook-censoredplanet.derived.merged_net_as`
- `firehook-censoredplanet.derived.merged_reduced_scans_no_as`

New view
- `firehook-censoredplanet.derived.merged_reduced_scans`

These will replace the old `derived.https*` tables, and are 3-4x larger than those tables, but are top-level clustered on the new `source` field which should help mitigate query costs.

demo dashboard: go/cp-explorer-2-laplante
short cost comparison: go/cp-explorer-2-laplante-cost-comparison

### CODE REVIEW

File: table/queries/merged_reduced_scans.sql [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    if(sent != "", TRIM(REGEXP_EXTRACT(sent, "Host: (.*)")), domain) as domain,
+    "DISCARD" AS source,
+    country,
+    netblock,
+    CleanError(error) AS result,
+    count(1) AS count
+  FROM `firehook-censoredplanet.base.discard_scan`
+  WHERE
+    error IS NULL OR
+    NOT SAFE.REGEXP_CONTAINS(error, "too many open files|address already in use|no route to host|connection refused|connect: connection timed out")
```
Comment (fortuna) [inline table/queries/merged_reduced_scans.sql:58]: Not for this PR, but I want to eventually stop dropping the data, and add an enum indicating what failed instead, so we can use the data for internet shutdowns. "too many open files" and "address already in use" will probably never be used, but changes to no route, connection refused, timeout can indicate a shutdown.

File: table/queries/merged_reduced_scans.sql [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    SELECT * FROM `firehook-censoredplanet.base.discard_scan` UNION ALL
+    SELECT * FROM `firehook-censoredplanet.base.echo_scan` UNION ALL
+    SELECT * FROM `firehook-censoredplanet.base.http_scan` UNION ALL
+    SELECT * FROM `firehook-censoredplanet.base.https_scan`
+  )
+);
+
+CREATE OR REPLACE TABLE `firehook-censoredplanet.derived.merged_reduced_scans_no_as`
+PARTITION BY date
+CLUSTER BY source, country, domain, netblock
```
Comment (fortuna) [inline table/queries/merged_reduced_scans.sql:43]: Is the source enough to limit the data? I was wondering if clustering works, or whether we needed separate tables in order to limit the data read.

**Review by fortuna — APPROVED** [review]

File: table/queries/merged_reduced_scans.sql (hunk 1 above)
Comment (ohnorobo) [reply table/queries/merged_reduced_scans.sql:58]: sounds good

File: table/queries/merged_reduced_scans.sql (hunk 2 above)
Comment (ohnorobo) [reply table/queries/merged_reduced_scans.sql:43]: The cost comparison from go/cp-explorer-2-laplante-cost-comparison shows that queries on the combined table are somewhat more expensive than using 1 or 4 different tables would be, but the clustering does seem to help somewhat.

Is there a method for having different tables in datastudio where the selected table is switched out by a selector? I don't know of one.

**Pull request merged**